include/ginkgo/core/matrix/batch_dense.hpp

```diff
@@ -338,6 +349,21 @@ class Dense final : public EnableBatchLinOp<Dense<ValueType>>,
 };
 
 
+/**
+ * Performs in-place row and column scaling for a given matrix.
+ *
+ * @param col_scale  the column scalars
+ * @param row_scale  the row scalars
+ * @param in_out  the matrix to be scaled
+ *
+ * @note the operation is performed in-place
+ */
+template <typename ValueType>
+void two_sided_scale(const array<ValueType>& col_scale,
+                     const array<ValueType>& row_scale,
+                     batch::matrix::Dense<ValueType>* in_out);
```

Comment:
ptr_param?

Reply:
I dont think there is a particular advantage in this case of `ptr_param` ?

Reply:
I second Mike's suggestion. This should use `ptr_param` since it doesn't involve ownership.